src/document_basic.c

```diff
@@ -351,11 +351,16 @@ void Document_Clear(Document *d) {
           rm_free(field->strval);
           break;
         case FLD_VAR_T_ARRAY:
-          for (int i = 0; i < field->arrayLen; ++i) {
-            rm_free(field->multiVal[i]);
```

Comment:
Before we freed the `multiVal` anyway, now there is a case where we do not free it at all, is that OK?

Reply:
We free it in all currently supported types for array: TEXT, TAG and now NUMERIC (in the else clause)
TEXT and TAG use field `multiVal` and NUMERIC use field `arrNumval`

Reply:
Can we reach this code and not enter one of those conditions?

Reply:
No, if we reach here - we enter one of those conditions.
We use unionType `FLD_VAR_T_ARRAY` only with TEXT TAG and NUMERIC
(VECTOR is using `FLD_VAR_T_CSTR` - so not reaching here)

We can add an assert.